src/components/application_manager/src/commands/mobile/unsubscribe_button_request.cc

```diff
@@ -41,13 +43,30 @@ namespace commands {
 
 namespace str = strings;
 
+namespace {
+struct FindButtonName {
+  FindButtonName(const mobile_apis::ButtonName::eType target_name)
+      : target_name_(target_name) {}
+
+  const mobile_apis::ButtonName::eType target_name_;
+
+  bool operator()(const smart_objects::SmartObject& buttonSO) const {
+    using namespace mobile_apis;
+    const ButtonName::eType name = static_cast<ButtonName::eType>(
+        buttonSO.getElement(hmi_response::button_name).asInt());
```

Comment:
@tkireva, actually, there should be `asInt()`, because `ButtonName` enum has a `INVALID_ENUM` value of which equal to -1